### PR TITLE
chore: KubernetesResourceUtilTest refactor

### LINKED ISSUE #3232: KubernetesResourceUtilTest 'initContainersInSpec.size() == 0' can be replaced with 'initContainersInSpec.isEmpty()'
### Description

The call to `List.size` can be replaced with `isEmpty` in [KubernetesResourceUtilTest](https://github.com/eclipse-jkube/jkube/blob/56c80d92bf28e7dcd8f7b63235758e02dd9397d9/jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtilTest.java).

The following line:
https://github.com/eclipse-jkube/jkube/blob/56c80d92bf28e7dcd8f7b63235758e02dd9397d9/jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtilTest.java#L632

should be changed to:

```java
    if (initContainersInSpec.isEmpty()) {
```

### :red_circle: Before you start :red_circle:

:point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down:

Make sure you read the [contributing guide](https://www.eclipse.org/jkube/contributing) first.

Pay special attention to the ECA agreement section and the requirement to sign-off your commit.

This is a [first-timers-only](https://eclipse.dev/jkube/contributing/#first-timers-only-issues) issue. These issues are especially created for contributors who are contributing to the project for the **first time**.

If you have already contributed to this project in the past, please don't ask to be assigned one of these issues and look for [good-first-issues](https://github.com/eclipse-jkube/jkube/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22) or [help-wanted](https://github.com/eclipse-jkube/jkube/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22help+wanted%22+) issues instead in order to continue you contributor journey in the project.

### PR DESCRIPTION
**Pull request by Piotr-Filochowski**
KubernetesResourceUtilTest refactor

## Description

KubernetesResourceUtilTest initContainersInSpec.size() == 0' replaced with 'initContainersInSpec.isEmpty()
https://github.com/eclipse-jkube/jkube/issues/3232

Fixes #3232

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3251 (_2024-07-21T15:00:41Z_)

:gear: [JKube E2E Tests (10029262547)](https://github.com/eclipse-jkube/ci/actions/runs/10029262547) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3251)](https://github.com/eclipse-jkube/ci/actions/runs/10029262547/job/27717230249)

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,8 @@ Usage:
 * Fix #2467: Add support for specifying imagePullSecrets via resource configuration
 * Fix #3220: ImageEnricher#mergeEnvVariables causes error for empty env
 * Fix #3228: Springboot 3.3.1 layertools output format breaks LayeredJarGenerator
+* Fix #3232: KubernetesResourceUtilTest refactor
```
Comment (rohanKanojia) [inline CHANGELOG.md:48]: Could you please elaborate on this line a bit? CHangelog is written for the users to get an idea of what's new in this specific release.

Comment (manusa) [reply CHANGELOG.md:48]: This is not adding any feature or change to production code in the project.
It's just a refactor fix, there's no changelog entry needed, it's better and less confusing once this is published as part of the release notes to remove this entry.

**Comment by Piotr-Filochowski** [comment]
@rohanKanojia @manusa Okey, line removed.

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**